src/v/cluster/health_monitor_backend.cc

```diff
@@ -329,48 +349,89 @@ ss::future<std::error_code>
 health_monitor_backend::dispatch_refresh_cluster_health_request(
   model::node_id node_id) {
     const auto timeout = model::timeout_clock::now() + max_metadata_age();
-    auto reply = co_await _connections.local()
-                   .with_node_client<controller_client_protocol>(
-                     _raft0->self().id(),
-                     ss::this_shard_id(),
-                     node_id,
-                     max_metadata_age(),
-                     [timeout](controller_client_protocol client) mutable {
-                         get_cluster_health_request req{
-                           .filter = cluster_report_filter{}};
-                         return client.get_cluster_health_report(
-                           std::move(req), rpc::client_opts(timeout));
-                     })
-                   .then(&rpc::get_ctx_data<get_cluster_health_reply>);
+    const auto self = _raft0->self().id();
+    /**
+     * Request remote node cluster health reports
+     */
+    auto request_f
+      = _connections.local()
+          .with_node_client<controller_client_protocol>(
+            self,
+            ss::this_shard_id(),
+            node_id,
+            max_metadata_age(),
+            [self, timeout](controller_client_protocol client) mutable {
+                get_cluster_health_request req{
+                  .filter = cluster_report_filter{.excluded_nodes = {self}}};
+                return client.get_cluster_health_report(
+                  std::move(req), rpc::client_opts(timeout));
+            })
+          .then(&rpc::get_ctx_data<get_cluster_health_reply>);
+    /**
+     * Instead of requesting current node report from the remote node, simply
+     * collect it using local state.
+     */
+    auto collect_f = collect_current_node_health(node_report_filter{});
```

Comment:
So we still need to do this because presumably the information locally moves from one subsystem to another?

Reply:
exactly